pkg/dispatcher/workers/controller.go

```diff
@@ -142,6 +143,16 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logge
 		return fmt.Errorf("couldn't get a list of calculations: %v", err)
 	}
 
+	workerPool := &workerpool.WorkerPool{}
+	err = r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: req.Namespace, Name: req.Name}, workerPool)
+	if err != nil && !kerrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get worker pool: %s in namespace %s: %w", req.Name, req.Namespace, err)
+	}
+	if workerPoolState, ok := workerPool.Spec.Workers[workerpool.WorkersName]; ok {
+		workerPoolState.State = workerpool.WorkerAvailableState
+	}
+	r.client.Update(ctx, workerPool)
```

Comment:
We need to check the error here. To avoid any revision conflicts, we need to retry to get and update again. Check https://github.com/vega-project/ccb-operator/blob/master/pkg/dispatcher/calculations/controller.go#L155-L170